workflows/rnaaln.nf

```diff
@@ -0,0 +1,846 @@
+include { CLEANUP as CLEAN_ALN_H} from '../modules/icgc-argo-workflows/cleanup/main'
+include { CLEANUP as CLEAN_ALN_S} from '../modules/icgc-argo-workflows/cleanup/main'
+include { CLEANUP as CLEAN_ALN_ST} from '../modules/icgc-argo-workflows/cleanup/main'
+
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    RUN MAIN WORKFLOW
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+workflow RNAALN {
+
+    ch_versions = Channel.empty()
+
+    // Validate input, generate metadata, prepare fastq channel
+    STAGE_INPUT(
+        params.study_id,
+        params.analysis_id,
+        params.samplesheet
+        )
+
+    ch_versions = ch_versions.mix(STAGE_INPUT.out.versions)
+
+    // Prepare reference file [meta fasta] [meta, fai]
+    ch_ref = Channel.fromPath(params.reference_fasta)
+                            .map{ path -> [ [id: 'fasta'], path ] }
+                            .mix( Channel.fromPath(params.reference_fai)
+                            .map{ path -> [ [id: 'fai'], path ] } )
+
+    // HISAT2 //
+    if (params.tools.split(',').contains('hisat2_aln')){
+
+        // HISAT2 - ALIGN //
+        index = Channel.fromPath(params.hisat2_index).collect()
+                .map { path -> [ [id: 'index'], path ] }
+
+        splicesites = Channel.fromPath(params.reference_splicesites).collect()
+                    .map { path -> [ [id: 'splicesites'], path ] }
+
+        // ALN in read group level
+        HISAT2_ALIGN(
+            STAGE_INPUT.out.meta_files,
+            index,
+            splicesites
+        )
+        ch_versions = ch_versions.mix(HISAT2_ALIGN.out.versions)
+
+        // MERG in sample level
+        MERG_DUP_H( //[val(meta), path(file1)],[[val(meta),[path(fileA)],[val(meta),[path(fileB)],]
+            HISAT2_ALIGN.out.bam,
+            ch_ref
+        )
+        ch_versions = ch_versions.mix(MERG_DUP_H.out.versions)
+
+        // Combine channels to determine upload status and payload creation
+        MERG_DUP_H.out.cram_alignment_index
+        .combine(STAGE_INPUT.out.upRdpc)
+        .combine(STAGE_INPUT.out.meta_analysis)
+        .map{
+            meta,cram,crai,upRdpc,metaB,analysis ->
+            [
+                [
+                    id:"${meta.study_id}.${meta.patient}.${meta.sample}.${meta.experiment}",
+                    patient:"${meta.patient}",
+                    sex:"${meta.sex}",
+                    sample:"${meta.sample}",
+                    read_group:"${meta.read_group}",
+                    data_type:"${meta.data_type}",
+                    date : "${meta.date}",
+                    genome_build: "${params.genome_build}",
+                    genome_annotation: "${params.genome_annotation}",
+                    read_groups_count: "${meta.numLanes}",
+                    study_id : "${meta.study_id}",
+                    date :"${new Date().format("yyyyMMdd")}",
```

Comment:
Defined twice? Ideally date should be set prior to payload generation. If it's used before then we run the risk of a workflow terminating and duplicate work being generated b/c of a new date variable.